### ISSUE TITLE
Cannot send config message to NEO-6 module

### ISSUE THREAD

**Issue by Bgoczan** [legokor/UPRA_OBC.stm32#13]
__SUMMARY:__ Cannot send config message to NEO-6 module
---
__Detailed Description:__

There is some problem in the command sending function:

```
void sendUBX(uint8_t* MSG, int len)
{
	int i=0;

	HAL_UART_Abort(&huart2);
	if (HAL_UART_Transmit(&huart2, (uint8_t*)(0xFF), 1, 500) != HAL_OK)
	{
		sendDebug("this shall not be seen");
		return;
	}

	HAL_Delay(500);
//	delay(5);
	for(i=0; i<len; i++)
	{
		//The problem might be here:
                if (HAL_UART_Transmit(&huart2, (uint8_t*)MSG[i], 1, 500) != HAL_OK)
		{
			sendDebug("this shall not be seen either");
			return;
		}
//	HAL_UART_Transmit(&huart2, (uint8_t*)MSG, strlen((char*)MSG), 500);

//		USART_SendData(USART2, MSG[i]);
	}
}
```
Warning message:
```
../Src/gps.c:442:34: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
   if (HAL_UART_Transmit(&huart2, (uint8_t*)MSG[i], 1, 500) != HAL_OK)
                                  ^
```

---

Branch: GPS_dev

Version: 60058a75c42a449bb019d22b05cf78bf339a424b

---

__Steps to Reproduce__
1. Connect uBlox NEO-6M module
2. Startup device wait for GPS init

---

Related Issues: #8

**Comment by martonbognar** [comment]
@Bgoczan something doesn't seem right here:

the parameter is `uint8_t* MSG`, meaning an array of `uint8_t` values.
if we want to send one character of this array (`MSG[i]`), we need the address of this character (`&MSG[i]`) as `HAL_UART_Transmit` takes a pointer as the second argument.

i suggest sending the address of the character, or sending the entire array in one go: `HAL_UART_Transmit(&huart2, (uint8_t*) MSG, len, 500)`.

**Comment by Bgoczan** [comment]
@martonbognar Thanks! I'll check it in the next coding session. It was a copy-paste code-snippet from the previous firmware version which used STM's Standard Peripheral Library and not the HAL libraries. 

This is low on the priority list. Issue #12 is a bigger concern right now.

**Comment by Bgoczan** [comment]
Fixed in 1dc75516eb8b08de3e56f96701d131ef066eb652
